internal/impl/devbox.go

```diff
@@ -384,48 +391,31 @@ func (d *Devbox) GenerateEnvrc(force bool, source string) error {
 		)
 	}
 	// confirm .envrc doesn't exist and don't overwrite an existing .envrc
-	if cmdutil.Exists("direnv") {
-		// prompt for direnv allow
-		var result string
-		isInteractiveMode := isatty.IsTerminal(os.Stdin.Fd())
-		if isInteractiveMode {
-			prompt := &survey.Input{
-				Message: "Do you want to enable direnv integration for this devbox project? [y/N]",
-			}
-			err := survey.AskOne(prompt, &result)
-			if err != nil {
-				return errors.WithStack(err)
-			}
-		}
-
-		if strings.ToLower(result) == "y" || !isInteractiveMode || source == "generate" {
-			if err := nix.EnsureNixInstalled(
-				d.writer, func() *bool { return lo.ToPtr(false) },
-			); err != nil {
-				return err
-			}
-
-			// generate all shell files to ensure we can refer to them in the .envrc script
-			if err := d.ensurePackagesAreInstalled(ctx, ensure); err != nil {
-				return err
-			}
+	if err := nix.EnsureNixInstalled(
+		d.writer, func() *bool { return lo.ToPtr(false) },
+	); err != nil {
+		return err
+	}
 
-			// .envrc file creation
-			err := generate.CreateEnvrc(tmplFS, d.projectDir)
-			if err != nil {
-				return errors.WithStack(err)
-			}
-		}
+	// generate all shell files to ensure we can refer to them in the .envrc script
+	if err := d.ensurePackagesAreInstalled(ctx, ensure); err != nil {
+		return err
+	}
 
-		if strings.ToLower(result) == "y" || !isInteractiveMode {
-			cmd := exec.Command("direnv", "allow")
-			err := cmd.Run()
-			if err != nil {
-				return errors.WithStack(err)
-			}
+	// .envrc file creation
+	err := generate.CreateEnvrc(tmplFS, d.projectDir)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	ux.Fsuccess(d.writer, "generated .envrc file\n")
```

Comment:
This should go in if statement right? We should not create `.envrc` file if user does not have direnv installed?

Reply:
@mikeland86 No we only check for direnv being installed to run `direnv allow`. I don't think it's necessary to check direnv being installed and prevent user from generating the file. They might have the direnv binary compiled and not in path or they might be generating the file for cicd or other other environments. With that logic we should also check if user has Docker installed before for `devbox generate dockerfile`.